core/eolearn/tests/test_core_tasks.py

```diff
@@ -148,28 +149,68 @@ def test_io_task_pickling(filesystem, task_class):
     assert isinstance(unpickled_task, task_class)
 
 
-def test_add_rename_remove_feature(patch):
+def test_add_feature(patch: EOPatch) -> None:
     cloud_mask = np.arange(10).reshape(5, 2, 1, 1)
-    feature_name = "CLOUD MASK"
-    new_feature_name = "CLM"
+    feature = (FeatureType.MASK, "CLOUD MASK")
 
-    patch = copy.deepcopy(patch)
+    with pytest.raises(KeyError):
+        patch[feature]
 
-    patch = AddFeatureTask((FeatureType.MASK, feature_name))(patch, cloud_mask)
-    assert np.array_equal(patch.mask[feature_name], cloud_mask), "Feature was not added"
+    patch = AddFeatureTask(feature)(patch, cloud_mask)
+    assert np.array_equal(patch[feature], cloud_mask)
 
-    patch = RenameFeatureTask((FeatureType.MASK, feature_name, new_feature_name))(patch)
-    assert np.array_equal(patch.mask[new_feature_name], cloud_mask), "Feature was not renamed"
-    assert feature_name not in patch[FeatureType.MASK], "Old feature still exists"
 
-    patch = RemoveFeatureTask((FeatureType.MASK, new_feature_name))(patch)
-    assert feature_name not in patch.mask, "Feature was not removed"
+@pytest.mark.parametrize(
+    "feature, feature_data",
+    [
+        ((FeatureType.BBOX, None), BBox((24.54, 56.45, 95.4, 13.43), CRS(3857))),
+        ((FeatureType.TIMESTAMP, None), [datetime(2022, 1, 1, 10, 4, 7), datetime(2022, 1, 4, 10, 14, 5)]),
+    ],
+)
+def test_add_bbox_timestamps(feature: FeatureSpec, feature_data: np.ndarray) -> None:
+    # this test should fail for bbox and timestamps after rework
+    patch = EOPatch()
+    assert not patch[feature]
+    patch = AddFeatureTask(feature)(patch, feature_data)
+    assert patch[feature] == feature_data
+
+
+def test_rename_feature(patch: EOPatch) -> None:
+    f_type, f_name, f_new_name = FeatureType.DATA, "bands", "new_bands"
+
+    with pytest.raises(KeyError):
+        patch[(f_type, f_name)]
+    patch_copy = copy.deepcopy(patch)
+
+    patch = RenameFeatureTask((f_type, f_name, f_new_name))(patch)
+    assert np.array_equal(patch[(f_type, f_new_name)], patch_copy[(f_type, f_name)])
+    assert f_name not in patch[f_type]
+
+
+def test_remove_feature(patch: EOPatch) -> None:
+    feature = (FeatureType.DATA, "bands")
+    patch_copy = copy.deepcopy(patch)
+    assert len(patch[feature]) != 0
```

Comment:
what are you really trying to test here exactly? It's not so clear from the assertion itself.

Reply:
usual we can test with assert object if something  exist (is not None) u can not do this kind of test for np.array this test is not the best so i m open for code improvement :)

Reply:
if the idea is only to test if the feature exists, you could just check with `assert feature in patch`, right?

Reply:
Yes, tnx done.